src/thriveopedia/Thriveopedia.cs

```diff
@@ -0,0 +1,338 @@
+
+            // Add all pages associated with a game in progress
+            if (currentGame != null)
+            {
+                AddPage("CurrentWorld");
+                AddPage("PatchMap", "CurrentWorld");
+                AddPage("EvolutionaryTree", "CurrentWorld");
+            }
+
+            // Notify all pages of the new game properties
+            foreach (var page in allPages.Keys)
+                page.CurrentGame = currentGame;
+        }
+    }
+
+    public override void _Ready()
+    {
+        backButton = GetNode<TextureButton>(BackButtonPath);
+        forwardButton = GetNode<TextureButton>(ForwardButtonPath);
+        pageContainer = GetNode<MarginContainer>(PageContainerPath);
+        pageTreeContainer = GetNode<PanelContainer>(PageTreeContainerPath);
+        pageTitle = GetNode<Label>(PageTitlePath);
+        pageTree = GetNode<Tree>(PageTreePath);
+
+        // Create and hide a blank root to avoid home being used as the root
+        pageTree.CreateItem();
+        pageTree.HideRoot = true;
+
+        // Keep a special reference to the home page
+        homePage = GetNode<ThriveopediaHomePage>(HomePagePath);
+        allPages.Add(homePage, CreateTreeItem(homePage, null));
+
+        // Add all pages not associated with a game in progress
+        AddPage("Museum");
+
+        pageHistory.Push(homePage);
+        SelectedPage = homePage;
+    }
+
+    /// <summary>
+    ///   Opens the Thriveopedia from the main menu.
+    /// </summary>
+    public void OpenFromMainMenu()
+    {
+        // Shouldn't do anything if already open
+        if (Visible)
+            return;
+
+        Show();
+    }
+
+    /// <summary>
+    ///   Opens the Thriveopedia from a particular game.
+    /// </summary>
+    public void OpenInGame(GameProperties game)
+    {
+        // Shouldn't do anything if already open
+        if (Visible)
+            return;
+
+        CurrentGame = game;
+
+        Show();
+    }
+
+    [RunOnKeyDown("ui_cancel", Priority = Constants.SUBMENU_CANCEL_PRIORITY)]
+    public bool OnEscapePressed()
+    {
+        // Only handle keypress when visible
+        if (!Visible)
+            return false;
+
+        Exit();
+
+        // If opened in a game, hide the pause menu too
+        return currentGame == null;
+    }
+
+    /// <summary>
+    ///   Opens an existing Thriveopedia page and adds it to the page history.
+    /// </summary>
+    /// <param name="pageName">The name of the page</param>
+    public void ChangePage(string pageName)
+    {
+        // By default, assume we're navigating to this page normally
+        ChangePage(pageName, true, true);
+    }
+
+    /// <summary>
+    ///   Gets an existing page by name, or null if no page exists with that name.
+    /// </summary>
+    /// <param name="name">The name of the desired page</param>
+    /// <returns>The Thriveopedia page with the given name</returns>
+    private ThriveopediaPage? GetPage(string? name)
```

Comment:
Why can the name to look for be null?

Reply:
If a page is created with no parent, this method gets called with null and returns null. I suppose when you put it like that it's better to avoid this path and check for null earlier.